e2e_correctness/query_neo_mem.py

```diff
@@ -58,6 +58,14 @@ def __eq__(self, other):
             if v != other._properties[k]:
                 logger.debug(f"Value {v} not equal to {other._properties[k]}")
                 return False
+            
+        for k, v in other._properties.items():
+            if k not in self._properties:
+                logger.debug(f"Property with key {k} not in {self._properties.keys()}")
+                return False
+            if v != self._properties[k]: #this if might be uselles, comment in review if you agree, this was checked in for loop above
+                logger.debug(f"Value {v} not equal to {self._properties[k]}")
```

Comment:
Generally I think we can do following:
1. find bigger map
2. iterate over that map

1. and 2. will cover if maps are equal and something is wrong. 

Add 3. check where we just check size of maps. If maps are of different size, no need to check again, just return false. Do you agree?